src/pages/Book.jsx

```diff
@@ -1,15 +1,92 @@
-import { Link, useLocation } from "react-router-dom";
+import { Link, useLocation, useNavigate, useParams } from "react-router-dom";
+import axios from "axios";
 import "../styles/Book.css";
+import { useEffect, useState } from "react";
 
 function Book() {
+  const { id } = useParams();
   const location = useLocation();
   const { book } = location.state;
+  const [chapters, setChapters] = useState([]);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const getListOfChapters = async () => {
```

Comment:
Here, like in the `BooksList`, you can define this function either outside, with the book id as a parameter, or inside the functional component, but ideally with a `useCallback` to memoize it.